src/query.ts

```diff
@@ -5,7 +5,12 @@ import {
   encodeQueryValue,
 } from "./encoding";
 
-export type QueryValue = string | undefined | null;
+export type QueryValue =
+  | string
+  | number
+  | undefined
+  | null
+  | Record<string, any>;
```

Comment:
should we also support arrays?

Reply:
Arrays are supported at function type level

Reply:
added tests too. ref for function: https://github.com/unjs/ufo/blob/4439503f650ad12aa320e5bbdf00f6dce75c2ce7/src/query.ts#L46